core/pom.xml

```diff
@@ -69,6 +69,30 @@
                 <configuration>
                     <shadedArtifactAttached>true</shadedArtifactAttached>
                     <shadedClassifierName>shaded</shadedClassifierName>
+                    <artifactSet>
+                        <excludes>
```

Comment:
What if we make the dependencies on the reader projects `provided` and `optional`?

I'd just include a comment explaining why we did it so that it's not accidentally removed late:

    <scope>provided</scope> <!-- because the shade plugin doesn't exclude optional dependencies -->